editor/connections_dialog.cpp

```diff
@@ -719,6 +719,17 @@ void ConnectDialog::_advanced_pressed() {
 	popup_centered();
 }
 
+void ConnectDialog::_tree_gui_input(const Ref<InputEvent> &p_event) {
+	Ref<InputEventMouseButton> mb = p_event;
+	if (mb.is_valid() && mb->get_button_index() == MouseButton::LEFT && mb->is_double_click() && mb->is_pressed()) {
+		Tree *scene_tree = tree->get_scene_tree();
+		if (scene_tree->get_pressed_button() == -1) {
+			return;
+		}
+		scene_tree->emit_signal(SNAME("item_activated"));
```

Comment:
```suggestion
		if (scene_tree->get_pressed_button() != -1) {
			scene_tree->emit_signal(SNAME("item_activated"));
		}
```
No reason for early return if it makes the code actually longer.